File: src/app/models/game-night.model.ts
import {Member} from './member.model';
import {Auth} from './auth.model';
import {Game} from './game.model';
import {Match} from './match.model';
import {Activity} from './activity.model';

interface IGameNight {
  GameNightId: string;
  Name: string;
  Members: Member[];
  Games: Game[];
  Matches: Match[];
  Notifications: Activity[];
  DateCreated: Date;
}

export class GameNight {

  public GameNightId: string;
  public Name: string;
  public Members: Member[];
  public Games: Game[];
  public Matches: Match[];
  public Notifications: Activity[];

  public DateCreated: Date;

  constructor(obj?: IGameNight) {
    this.GameNightId = obj && obj.GameNightId || "";
    this.Name = obj && obj.Name || "";
    this.Members = obj && obj.Members || new Array<Member>();
    this.Games = obj && obj.Games || new Array<Game>();
    this.Matches = obj && obj.Matches || new Array<Match>();
    this.Notifications = obj && obj.Notifications || new Array<Activity>();
    this.DateCreated = obj && obj.DateCreated || new Date();
  }
  
}

File: GameNightBuddy-Server/src/GameNightBuddy-Server/Models/GameNight.Model.cs
using GameNightBuddy_Server.ViewModels;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace GameNightBuddy_Server.Models
{
  public class GameNight
  {
    // Primary Key
    public Guid GameNightId { get; set; }
    // Foreign Keys
    [Required]
    public Guid UserId { get; set; }
    public User UserCreator { get; set; }
    // Foreign Keys Elsewhere
    public List<GameNightMember> Members { get; set; }
    public List<GameNightGame> Games { get; set; }
    public List<Match> Matches { get; set; }
    public List<Activity> Activities { get; set; }

    // Unique Columns
    [Required]
    public string Name { get; set; } 

    // Default
    public DateTime DateCreated { get; set; }
    public string UserCreated { get; set; }
    public DateTime DateUpdated { get; set; }
    public string UserUpdated { get; set; }
    public Boolean IsActive { get; set; }

    public GameNight() { }

    public GameNight(GameNightViewModel vm)
    {
      Name = vm.Name;
    }
  }
}
